sda-download/api/sda/sda.go

```diff
@@ -29,6 +33,43 @@ func sanitizeString(str string) string {
 	return pattern.ReplaceAllString(str, "[identifier]: $1")
 }
 
+func reencryptHeader(oldHeader []byte, reencKey string) ([]byte, error) {
```

Comment:
This function is missing a test case

Reply:
Since this function is doing the connection of `download` to `reencrypt`, I think it is better to be tested in the integration test. As to the unit tests of the functionality of reencrypt, they are already fully covered at https://github.com/neicnordic/sensitive-data-archive/blob/main/sda/cmd/reencrypt/reencrypt_test.go.